Firestore/core/src/core/query.h

```diff
@@ -277,25 +277,34 @@ class Query {
   // sort at the end.
   std::vector<OrderBy> explicit_order_bys_;
 
-  // The memoized list of sort orders.
-  mutable std::vector<OrderBy> memoized_normalized_order_bys_;
-
   int32_t limit_ = Target::kNoLimit;
   LimitType limit_type_ = LimitType::None;
 
   absl::optional<Bound> start_at_;
   absl::optional<Bound> end_at_;
 
+  Target ToTarget(const std::vector<OrderBy>& order_bys) const;
+
+  // For properties below, use a `std::shared_ptr<ThreadSafeMemoizer>` rather
+  // than using `ThreadSafeMemoizer` directly so that this class is copyable
+  // (`ThreadSafeMemoizer` is not copyable because of its `std::once_flag`
+  // member variable, which is not copyable).
+
+  // The memoized list of sort orders.
+  mutable std::shared_ptr<ThreadSafeMemoizer<std::vector<OrderBy>>>
```

Comment:
nit: Add a `#include` for the file that declares `ThreadSafeMemoizer`. This satisfies the "include what you use" principal, where you shouldn't rely on transitive includes.